#!/usr/bin/env python

PACKAGE = 'bcontrol'

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, bool_t, double_t

gen = ParameterGenerator()

# Angular
gen.add("Kp_feedforward", double_t, 0, "Feedforward proportional gain.", 1.0, -10.0, 10.0)
gen.add("Kp_heading", double_t, 0, "Heading proportional gain.", 2.8, -10.0, 10.0)
gen.add("Kp_lateral_position", double_t, 0, "Lateral position proportional gain.", 10.0, -20.0, 20.0)
gen.add("Kp_angvel", double_t, 0, "Angular velocity proportional gain.", 3.4, -10.0, 10.0)

# Linear
gen.add("Kp_vel", double_t, 0, "Velocity proportional gain.", 1.0, -10.0, 10.0)

# Other parameters
gen.add("lookahead_m", double_t, 0, "Lookahead distance in meters.", -0.1, -10.0, 10.0)

exit(gen.generate(PACKAGE, "bcontrol", "BController"))
